### PR TITLE
[FIX]point_of_sale: Tax should also be apply when tax computation method is 'Group of Taxes'.

### PR DESCRIPTION
**Pull request by dka-odoo**
…p tax(Tax Computation = group of taxes).

Issue :- https://www.odoo.com/web#id=646306&view_type=form&model=project.issue&menu_id=4120&action=609

### CODE REVIEW

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -211,7 +211,7 @@ exports.PosModel = Backbone.Model.extend({
             });
             _.each(self.taxes_by_id, function(tax) {
                 tax.child_taxes = {};
-                _.each(tax.child_ids, function(child_tax_id) {
+                _.each(tax.children_tax_ids, function(child_tax_id) {
                     tax.child_taxes[child_tax_id] = self.taxes_by_id[child_tax_id];
                 });
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/js/models.js]: We should try and keep the POS accounting stuff as consistent as possible with that of `account.py`. It's already confusing enough as it is. So a tax in the POS should have `children_taxes_ids` that is an array of other account.tax references. So essentially all you need to do is replace the array of ids with references to the correct taxes. Something like:

```javascript
_.each(self.taxes_by_id, function(tax) {
    // children_tax_ids is a m2m, which get loaded as an
    // array of ids, this just replaces the ids with the
    // corresponding account.tax references
    tax.children_tax_ids = _.map(tax.children_tax_ids, function (child_tax_id) {
        return self.taxes_by_id[child_tax_id];
    });
});
```

would be nice.

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -1347,11 +1347,11 @@ exports.Orderline = Backbone.Model.extend({
         _(taxes).each(function(tax) {
             tax = self._map_tax_fiscal_position(tax);
             if (tax.amount_type === 'group'){
-                var ret = self.compute_all(tax.children_tax_ids, price_unit, quantity, currency_rounding);
+                var ret = self.compute_all(tax.child_taxes, price_unit, quantity, currency_rounding);
```
Comment (jorenvo) [inline addons/point_of_sale/static/src/js/models.js]: See above. This change can be discarded.

File: addons/point_of_sale/static/src/js/models.js (hunk 1 above)
Comment (dka-odoo) [inline addons/point_of_sale/static/src/js/models.js]: @jorenvo  you are right, we can remove the child_taxes :)

**Comment by jorenvo** [comment]
Thanks, merged in 1b714bdc6ce75f856509548f2190d4e643186d58.

**Pull request closed without merging**